__tests__/transform.spec.js

```diff
@@ -91,3 +91,11 @@ defineTest(
   { "preserve-prop-types": "unconverted" },
   "preserve-unconverted-shape"
 )
+
+// --declaration-style=type
+defineTest(
+  __dirname,
+  transform,
+  { "declaration-style": "type" },
+  "declaration-style-type"
+)
```

Comment:
Way less tests than in the first draft ;)
LMK if you'd like some additional ones

Reply:
Nope, this seems fine.